### ISSUE TITLE
Allow upload of gpx file

### ISSUE THREAD

**Issue by westis** [TrailRouter/public-issues#34]
MAny times I may have a gpx file that I want to use as a starting point or to have Trail Router "analyze it" for elevation, greenery, way and surface types etc.

Would be nice if I could upload my own gpx file, although this is likely not the primary use for TR.

**Comment by samcrawford** [comment]
I've thought about this too, as I think it'd be useful for me, but it'd have a lot of limitations. Once a GPX was uploaded, you couldn't then add waypoints to it or modify the route in any way - this is because our routing decisions will not match whatever was used to create the original GPX file, so adding one waypoint would cause the whole route to be recalculated, and that would confuse users greatly.

Maybe the answer is to allow people to add in GPX files as additional read-only layers?

**Comment by westis** [comment]
Yeah, unless some waypoints are created on import? That seems to be how Komoot does it and when dragging one waypoint, only the segment between the neareast waypoints is being recalculated, while keeping all the others intact.

Import of gpx files is probably not a high priority though, as Trail Router is mainly for creating routes from scratch. Other tools could be used to import and possibly edit an existing gpx file.
